dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -83,25 +81,45 @@ protected CommunityRest createAndReturn(Context context) throws AuthorizeExcepti
         }
 
         Community community;
+        try {
+            // top-level community
+            community = cs.create(null, context);
+            cs.update(context, community);
+            metadataConverter.setMetadata(context, community, communityRest.getMetadata());
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
 
+        return dsoConverter.convert(community);
+    }
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'COMMUNITY', 'ADD')")
+    protected CommunityRest createAndReturn(Context context, UUID id) throws AuthorizeException {
+
+        if (id == null) {
+            throw new DSpaceBadRequestException("The given parent parameter was invalid: "
+                + id);
```

Comment:
Similar to above, the `id` is known to be null.  Maybe reword this to be something like `"Parent Community UUID is null. Cannot create a SubCommunity without providing a parent Community"`